src/test/regress/expected/alter_table_set_access_method.out

```diff
@@ -681,6 +681,40 @@ CREATE TABLE identity_cols_test (a int, b int generated by default as identity (
 ERROR:  cannot complete command because relation alter_table_set_access_method.identity_cols_test has identity column
 HINT:  Drop the identity columns and re-try the command
+-- test long table names
+SET client_min_messages TO DEBUG1;
+CREATE TABLE abcde_0123456789012345678901234567890123456789012345678901234567890123456789 (x int, y int);
+NOTICE:  identifier "abcde_0123456789012345678901234567890123456789012345678901234567890123456789" will be truncated to "abcde_012345678901234567890123456789012345678901234567890123456"
+SELECT create_reference_table('abcde_0123456789012345678901234567890123456789012345678901234567890123456789');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT alter_table_set_access_method('abcde_0123456789012345678901234567890123456789012345678901234567890123456789', 'columnar');
+DEBUG:  the name of the shard (abcde_01234567890123456789012345678901234567890_f7ff6612_xxxxxx) for relation (abcde_012345678901234567890123456789012345678901234567890123456) is too long, switching to sequential and local execution mode to prevent self deadlocks
+NOTICE:  creating a new table for alter_table_set_access_method.abcde_012345678901234567890123456789012345678901234567890123456
+DEBUG:  pathlist hook for columnar table am
+CONTEXT:  SQL statement "SELECT EXISTS (SELECT 1 FROM alter_table_set_access_method.abcde_0123456789012345678901234567890123456_f7ff6612_4160710162)"
+DEBUG:  updating chunk row count to 10000
+DEBUG:  updating stripe row count to 150000
+DEBUG:  updating compression to zstd
+DEBUG:  updating compression level to 3
+NOTICE:  Moving the data of alter_table_set_access_method.abcde_012345678901234567890123456789012345678901234567890123456
+NOTICE:  Dropping the old alter_table_set_access_method.abcde_012345678901234567890123456789012345678901234567890123456
+CONTEXT:  SQL statement "DROP TABLE alter_table_set_access_method.abcde_012345678901234567890123456789012345678901234567890123456 CASCADE"
+CONTEXT:  SQL statement "SELECT citus_drop_all_shards(v_obj.objid, v_obj.schema_name, v_obj.object_name)"
+PL/pgSQL function citus_drop_trigger() line 16 at PERFORM
+SQL statement "DROP TABLE alter_table_set_access_method.abcde_012345678901234567890123456789012345678901234567890123456 CASCADE"
+NOTICE:  Renaming the new table to alter_table_set_access_method.abcde_012345678901234567890123456789012345678901234567890123456
+DEBUG:  the name of the shard (abcde_01234567890123456789012345678901234567890_f7ff6612_xxxxxx) for relation (abcde_012345678901234567890123456789012345678901234567890123456) is too long, switching to sequential and local execution mode to prevent self deadlocks
+CONTEXT:  SQL statement "ALTER TABLE alter_table_set_access_method.abcde_0123456789012345678901234567890123456_f7ff6612_4160710162 RENAME TO abcde_012345678901234567890123456789012345678901234567890123456"
+ alter_table_set_access_method
```

Comment:
Again, if we can do `SELECT * FROM table xxx` we can ensure that shards are in a good shape.

Reply:
Done